static/app/views/detectors/components/detectorLink.tsx

```diff
@@ -1,43 +1,183 @@
 import {makeMonitorDetailsPathname} from 'sentry/views/detectors/pathnames';
 
 type DetectorLinkProps = {
-  createdBy: string | null;
-  detectorId: string;
-  disabled: boolean;
-  name: string;
-  projectId: string;
+  detector: Detector;
   className?: string;
 };
 
-export function DetectorLink({
-  detectorId,
-  name,
-  createdBy,
-  projectId,
-  className,
-  disabled,
-}: DetectorLinkProps) {
+function formatConditionPriority(condition: DataCondition) {
+  switch (condition.conditionResult) {
+    case DetectorPriorityLevel.HIGH:
+      return 'high';
+    case DetectorPriorityLevel.MEDIUM:
+      return 'medium';
+    default:
+      return 'low';
+  }
+}
+
+function formatConditionType(condition: DataCondition) {
+  switch (condition.type) {
+    case DataConditionType.GREATER:
+      return '>';
+    case DataConditionType.LESS:
+      return '<';
+    case DataConditionType.EQUAL:
+      return '=';
+    case DataConditionType.NOT_EQUAL:
+      return '!=';
+    case DataConditionType.GREATER_OR_EQUAL:
+      return '>=';
+    case DataConditionType.LESS_OR_EQUAL:
+      return '<=';
+    default:
+      return condition.type;
+  }
+}
+
+function formatCondition({condition, unit}: {condition: DataCondition; unit: string}) {
+  if (
+    !condition.conditionResult ||
+    condition.conditionResult === DetectorPriorityLevel.OK
+  ) {
+    return null;
+  }
+
+  const comparison = formatConditionType(condition);
+  const threshold = `${condition.comparison}${unit}`;
+  const priority = formatConditionPriority(condition);
+
+  return `${comparison}${threshold} ${priority}`;
+}
+
+function DetailItem({children}: {children: React.ReactNode}) {
+  if (!children) {
+    return null;
+  }
+
+  return (
+    <Fragment>
+      <Separator />
+      <DetailItemContent>{children}</DetailItemContent>
+    </Fragment>
+  );
+}
+
+function ConfigDetails({detector}: {detector: Detector}) {
+  const type = detector.config.detection_type;
+  const conditions = detector.conditionGroup?.conditions;
+  if (!conditions?.length) {
+    return null;
+  }
+
+  switch (type) {
+    case 'static': {
+      const text = conditions
+        .map(condition => formatCondition({condition, unit: 's'}))
+        .filter(defined)
+        .join(', ');
+      if (!text) {
+        return null;
+      }
+      return <DetailItem>{text}</DetailItem>;
+    }
+    case 'percent': {
+      const text = conditions
+        .map(condition => formatCondition({condition, unit: '%'}))
+        .filter(defined)
+        .join(', ');
+      if (!text) {
+        return null;
+      }
+      return <DetailItem>{text}</DetailItem>;
+    }
+    case 'dynamic':
+      return <DetailItem>{t('Dynamic')}</DetailItem>;
+    default:
+      unreachable(type);
+      return null;
+  }
+}
+
+function DataSourceDetails({dataSource}: {dataSource: DataSource}) {
+  const type = dataSource.type;
+  switch (type) {
+    case 'snuba_query_subscription':
```

Comment:
similar nit as the other -- a lookup object might be a little easier to grok;

```ts
const DataSourceDetails = {
   "snuba_query_subscription": (
       <Fragment>
           ...
       </Fragment>
   )
}
```

Using that structure would allow us to also have specific typing to the intended data source per component:
```ts
const DataSourceDetails = {
   "snuba_query_subscription": ({ dataSource: SnubaQueryDataSource }) => (
       <Fragment>
           ...
       </Fragment>
   )
}
```